Nadim.CinemaReservationSystem.Web/Controllers/CinemaController.cs

```diff
@@ -1,58 +1,55 @@
-﻿using System;
-using System.Collections.Generic;
-using System.Linq;
-using System.Threading.Tasks;
+﻿using Microsoft.AspNetCore.Authorization;
 using Microsoft.AspNetCore.Mvc;
-using Nadim.CinemaReservationSystem.Web.Models;
 using Nadim.CinemaReservationSystem.Web.Contracts;
-using Microsoft.AspNetCore.Authorization;
+using Nadim.CinemaReservationSystem.Web.Models;
 
 namespace Nadim.CinemaReservationSystem.Web.Controllers
 {
-    [Route("api/[controller]")]
+    [Route("api/cinemas")]
     public class CinemaController : Controller
     {
-        private readonly ICinemaService CinemaService;
+        private readonly ICinemaService сinemaService;
 
-        public CinemaController(ICinemaService createCinemaService)
+        public CinemaController(ICinemaService сinemaService)
         {
-            this.CinemaService = createCinemaService;
+            this.сinemaService = сinemaService;
         }
 
         [Authorize(Roles = "admin")]
-        [HttpPost("[action]")]
-        public IActionResult AddCinema([FromBody] CinemaCreationInfo cinemaInfo)
+        [HttpPost]
+        public ActionResult<Result> AddCinema([FromBody] CinemaCreationInfo cinemaInfo)
         {
-            Result result = CinemaService.CreateCinema(cinemaInfo);
+            Result result = сinemaService.CreateCinema(cinemaInfo);
 
             if (result.ResultOk)
             {
-                return Ok(result);
+                return Created("api/cinemas/" + ((ResultCreated)result).Uri, typeof(Cinema));
```

Comment:
Don't use upcasting, return ResultCreated from service method.
Use string interpolation instead of string concatenation.